dill/session.py

```diff
@@ -101,36 +108,64 @@ def _restore_modules(unpickler, main_module):
     except KeyError:
         pass
 
-#NOTE: 06/03/15 renamed main_module to main
-def dump_session(filename='/tmp/session.pkl', main=None, byref=False, **kwds):
+def _filter_objects(main, exclude_extra, include_extra, obj=None):
+    filters = ExcludeRules(getattr(settings, 'session_exclude', None))
+    if exclude_extra is not None:
+        filters.update([(EXCLUDE, exclude_extra)])
+    if include_extra is not None:
+        filters.update([(INCLUDE, include_extra)])
+
+    namespace = filters.filter_namespace(vars(main), obj=obj)
+    if namespace is vars(main):
+        return main
+
+    main = ModuleType(main.__name__)
+    vars(main).update(namespace)
+    return main
+
+def dump_session(filename: Union[PathLike, BytesIO] = '/tmp/session.pkl',
+                 main: Union[str, ModuleType] = '__main__',
+                 byref: bool = False,
+                 exclude: Union[Filter, Iterable[Filter]] = None,
+                 include: Union[Filter, Iterable[Filter]] = None,
+                 **kwds) -> NoReturn:
     """pickle the current state of __main__ to a file"""
-    from .settings import settings
-    protocol = settings['protocol']
-    if main is None: main = _main_module
+    protocol = settings.protocol
+    if isinstance(main, str):
+        main = _import_module(main)
+    original_main = main
+    if byref:
+        #NOTE: *must* run before _filter_objects()
+        main = _stash_modules(main)
+    main = _filter_objects(main, exclude, include, obj=original_main)
+
+    print(list(vars(main)))
```

Comment:
I forgot to remove this line before committing, but it is removed in the next one.